easybuild/easyconfigs/f/FreeFem++/FreeFem++-3.60-intel-2018a-download-deps.eb

```diff
@@ -0,0 +1,45 @@
+easyblock = 'ConfigureMake'
+
+name = 'FreeFem++'
+version = '3.60'
+versionsuffix = '-downloaded-deps'
+
+homepage = 'http://www.freefem.org/'
+description = """ FreeFem++ is a partial differential equation solver. It has its own language.
+freefem scripts can solve multiphysics non linear systems in 2D and 3D. """
+
+toolchain = {'name': 'intel', 'version': '2018a'}
+toolchainopts = {'usempi': True, 'pic': True}
+
+source_urls = ['http://www.freefem.org/ff++/ftp/']
+sources = ['%(namelower)s-%(version)s.tar.gz']
+checksums = ['8acc37a89487d2b0db5088f3f8badda971bab1c4e8f64d24920d23890804901c']
+
+builddependencies = [('Autotools', '20170619')]
+
+dependencies = [
+    ('HDF5', '1.10.1'),
+    ('GSL', '2.4'),
+    ('freeglut', '3.0.0'),
+]
+
+preconfigopts = "autoreconf -i &&"
+
+configopts = '--with-hdf5="$EBROOTHDF5"/bin '
+configopts += '--with-gsl-include="$EBROOTGSL"/include --with-gsl-ldflags="$EBROOTGSL"/lib '
+configopts += '--with-lapack="$LIBLAPACK" '
+
+# dependencies are heavily patched by FreeFem++, we therefore let it download and install them itself
+prebuildopts = 'download/getall -a -o ScaLAPACK,ARPACK,freeYams,Gmm++,Hips,Ipopt,METIS,ParMETIS,MMG3D,mshmet,MUMPS,'
+prebuildopts += 'NLopt,pARMS,PaStiX,Scotch,SuiteSparse,SuperLU_DIST,SuperLU,TetGen,PETSc,SLEPc,hpddm &&'
+
+parallel = 1
+
+sanity_check_paths = {
+    'files': ['bin/%s' % x for x in ['bamg', 'cvmsh2', 'ffglut', 'ffmedit']] +
+             ['bin/ff-%s' % x for x in ['c++', 'get-dep', 'mpirun', 'pkg-download']] +
+             ['bin/FreeFem++%s' % x for x in ['', '-mpi', '-nw']],
+    'dirs': ['share/freefem++/3.60/'] + ['lib/ff++/3.60/%s' % x for x in ['bin', 'etc', 'idp', 'include', 'lib']]
```

Comment:
@SimonPinches Please use `%(version)s` rather than hardcoding `3.60`

Reply:
If I just substitute `%(version)s` it doesn't work (`Parsing easyconfig file failed: format requires a mapping (line 43)`), presumably because of the loop over `x` later.

Reply:
Yeah, you need to escape the template using `%%(version)s` (and then avoid the line becoming too long):

```python
'dirs': ['share/freefem++/%(version)s/'] +
        ['lib/ff++/%%(version)s/%s' % x for x in ['bin', 'etc', 'idp', 'include', 'lib']]

Reply:
See https://github.com/easybuilders/easybuild-easyconfigs/pull/5918/files